src/ec/eddsa.rs

```diff
@@ -171,11 +171,8 @@ pub static ED25519: EdDSAParameters = EdDSAParameters {};
 impl signature::VerificationAlgorithm for EdDSAParameters {
     fn verify(&self, public_key: untrusted::Input, msg: untrusted::Input,
               signature: untrusted::Input) -> Result<(), error::Unspecified> {
-        if public_key.len() != PUBLIC_KEY_LEN {
-            return Err(error::Unspecified);
-        }
         let public_key = public_key.as_slice_less_safe();
-        let public_key = slice_as_array_ref!(public_key, ELEM_LEN).unwrap();
+        let public_key = try!(slice_as_array_ref!(public_key, ELEM_LEN));
```

Comment:
Hmm, this looks much more idiomatic. The only thing I'm hesitant about is that, per the `check!` discussion, the branching— and more importantly, input validation— now feels hidden.

I know that we're using the length check of `slice_as_array_ref!` to validate the input length. But, the existence of this check is less obvious now. I guess if we were only _really_ checking `public_key.len()` to ensure our `unwrap()` could never fail, then this is an unambiguous improvement. But, if we also care about `public_key.len()` being a valid length for its own sake, then I'd prefer not to push that check into the `slice_as_array_ref!` macro. In short, I'd prefer input validation to be explicit, and right now it feels a little implicit.

What do you think?